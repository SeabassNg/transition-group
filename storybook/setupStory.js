import React from 'react'
import { Provider, connect } from 'react-redux'

import configureStore from './configureStore'
import { connectStore } from './connectStore'
import { AnimatedTransitionGroup, AnimatedChild } from '../src/index'

import './styles.css'


export default (groupProps, childProps) => {
  const store = configureStore()
  const TransitionGroup = transitionGroupHOC(groupProps, childProps)
  return connectStore(store, TransitionGroup)
}


export const transitionGroupHOC = (groupProps, childProps) => {
  const Wrapper = ({ num }) =>
    <AnimatedTransitionGroup component="div" duration={2500} {...groupProps}>
      <AnimatedChild key={num} {...childProps}>
        <div className='child' style={{ backgroundColor: color(num) }} />
      </AnimatedChild>
    </AnimatedTransitionGroup>

  return connect(state => state)(Wrapper)
}

const color = num =>
  num % 2 === 0 ? 'blue' : 'purple'

