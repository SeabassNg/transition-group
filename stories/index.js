import setupStory from '../storybook/setupStory'


jest.mock('../src/dom-utils', () => ({
  addClass: (component, className, name) => {
    expect({ method: 'addClass', className, name }).toMatchSnapshot()
  },
  removeAnimationClasses: (component, name) => {
    expect({ method: 'removeAnimationClasses', name }).toMatchSnapshot()
  },
  setTimeoutAnimationFrame: (func, ms, className) => {
    expect({ method: 'setTimeoutAnimationFrame', ms, className }).toMatchSnapshot()
    func()
  },
}))


describe('AnimatedTransitionGroup 1', () => {
  it('appear', () => {
    const app = setupStory()

    console.log(app.tree())
    console.log(app.tree().children[0])

    app.snap()

    return app.story()
  })

  it('enter', () => {
    const app = setupStory()

    app.dispatch({ type: 'SWITCH' })

    console.log(app.tree())
    console.log(app.tree().children[0])

    app.snap()

    return app.story()
  })
})

/**
describe('AnimatedTransitionGroup 2', () => {
  const { story, store } = setupStory()

  it('orange', () => {
    store.dispatch({ type: 'CHANGE', payload: 'orange' })
    store.dispatch({ type: 'BLUR', payload: 'orange' })

    const { color } = store.getState()
    expect(color).toEqual('orange')

    const component = renderer.create(story)
    expect(component).toMatchSnapshot()

    return story
  })

  it('green', () => {
    store.dispatch({ type: 'CHANGE', payload: 'green' })
    store.dispatch({ type: 'BLUR', payload: 'green' })

    const { color } = store.getState()
    expect(color).toEqual('green')

    const component = renderer.create(story)
    expect(component).toMatchSnapshot()

    return story
  })

  it('just tests, no story returned', () => {
    expect(1).toEqual(1)
    expect(2).toEqual(2)
    expect(3).toEqual(3)
    expect(1).toEqual(1)

    // notice no story is returned
  })

  it('more equal tests', () => {
    expect(66).toEqual(66)
    expect(55).toEqual(55)

    // notice no story is returned
  })
})
**/
