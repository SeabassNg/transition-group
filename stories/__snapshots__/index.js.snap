// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimatedTransitionGroup 1 appear 1`] = `
Object {
  "className": "appear",
  "method": "addClass",
  "name": "",
}
`;

exports[`AnimatedTransitionGroup 1 appear 2`] = `
Object {
  "className": "appear",
  "method": "setTimeoutAnimationFrame",
  "ms": 0,
}
`;

exports[`AnimatedTransitionGroup 1 appear 3`] = `
Object {
  "className": "appear-active",
  "method": "addClass",
  "name": "",
}
`;

exports[`AnimatedTransitionGroup 1 appear 4`] = `
Object {
  "className": "done",
  "method": "setTimeoutAnimationFrame",
  "ms": 1,
}
`;

exports[`AnimatedTransitionGroup 1 appear 5`] = `
Object {
  "method": "removeAnimationClasses",
  "name": undefined,
}
`;

exports[`AnimatedTransitionGroup 1 appear 6`] = `
<div>
  <div
    className="child"
    style={
      Object {
        "backgroundColor": "blue",
      }
    }
  />
</div>
`;

exports[`AnimatedTransitionGroup 1 enter 1`] = `
Object {
  "className": "appear",
  "method": "addClass",
  "name": "",
}
`;

exports[`AnimatedTransitionGroup 1 enter 2`] = `
Object {
  "className": "appear",
  "method": "setTimeoutAnimationFrame",
  "ms": 0,
}
`;

exports[`AnimatedTransitionGroup 1 enter 3`] = `
Object {
  "className": "appear-active",
  "method": "addClass",
  "name": "",
}
`;

exports[`AnimatedTransitionGroup 1 enter 4`] = `
Object {
  "className": "done",
  "method": "setTimeoutAnimationFrame",
  "ms": 1,
}
`;

exports[`AnimatedTransitionGroup 1 enter 5`] = `
Object {
  "method": "removeAnimationClasses",
  "name": undefined,
}
`;

exports[`AnimatedTransitionGroup 1 enter 6`] = `
Object {
  "className": "enter",
  "method": "addClass",
  "name": "",
}
`;

exports[`AnimatedTransitionGroup 1 enter 7`] = `
Object {
  "className": "enter",
  "method": "setTimeoutAnimationFrame",
  "ms": 0,
}
`;

exports[`AnimatedTransitionGroup 1 enter 8`] = `
Object {
  "className": "enter-active",
  "method": "addClass",
  "name": "",
}
`;

exports[`AnimatedTransitionGroup 1 enter 9`] = `
Object {
  "className": "done",
  "method": "setTimeoutAnimationFrame",
  "ms": 1,
}
`;

exports[`AnimatedTransitionGroup 1 enter 10`] = `
Object {
  "method": "removeAnimationClasses",
  "name": undefined,
}
`;

exports[`AnimatedTransitionGroup 1 enter 11`] = `
Object {
  "className": "leave",
  "method": "addClass",
  "name": "",
}
`;

exports[`AnimatedTransitionGroup 1 enter 12`] = `
Object {
  "className": "leave",
  "method": "setTimeoutAnimationFrame",
  "ms": 0,
}
`;

exports[`AnimatedTransitionGroup 1 enter 13`] = `
Object {
  "className": "leave-active",
  "method": "addClass",
  "name": "",
}
`;

exports[`AnimatedTransitionGroup 1 enter 14`] = `
Object {
  "className": "done",
  "method": "setTimeoutAnimationFrame",
  "ms": 1,
}
`;

exports[`AnimatedTransitionGroup 1 enter 15`] = `
<div>
  <div
    className="child"
    style={
      Object {
        "backgroundColor": "purple",
      }
    }
  />
</div>
`;
